packages/venia-concept/src/components/MiniCart/product.js

```diff
@@ -22,14 +26,26 @@ const Product = props => {
 
     const classes = mergeClasses(defaultClasses, props.classes);
     const mask = isLoading ? <div className={classes.mask} /> : null;
-    const imageSource = useMemo(
-        () =>
-            resourceUrl(image.file, {
-                type: 'image-product',
-                width: imageWidth
-            }),
-        [image]
-    );
+
+    const productImage = useMemo(() => {
+        const src =
+            image && image.file
+                ? resourceUrl(image.file, {
+                      type: 'image-product',
+                      width: imageWidth,
+                      height: imageHeight
+                  })
+                : transparentPlaceholder;
+
+        return (
+            <Image
+                alt={name}
+                classes={{ root: classes.image }}
+                placeholder={transparentPlaceholder}
+                src={src}
```

Comment:
I'm wondering if we shouldn't just have the `Image` component render the `placeholder` when the `src` image fails to load. We could get rid of the conditional logic above (L32) if the component would take care of that for us.

Reply:
It's already doing that. It'll render a placeholder _until_ the `src` loads, which means if it fails it'll just stay as the placeholder. What the conditional is doing is saying "if there is no source, just render a placeholder AS the source`.